packages/components/date-picker/src/date-picker-com/panel-month-range.vue

```diff
@@ -99,71 +99,47 @@
 </template>
 
 <script lang="ts" setup>
-import { computed, inject, ref, toRef, useAttrs, useSlots, watch } from 'vue'
+import { computed, inject, ref, toRef, watch } from 'vue'
 import dayjs from 'dayjs'
 import ElIcon from '@element-plus/components/icon'
-import { useLocale, useNamespace } from '@element-plus/hooks'
-import { isArray, isFunction } from '@element-plus/utils'
+import { useLocale } from '@element-plus/hooks'
+import { isArray } from '@element-plus/utils'
 import { DArrowLeft, DArrowRight } from '@element-plus/icons-vue'
-import { ROOT_PICKER_INJECTION_KEY } from '@element-plus/tokens'
 import {
   panelMonthRangeEmits,
   panelMonthRangeProps,
 } from '../props/panel-month-range'
 import { useMonthRangeHeader } from '../composables/use-month-range-header'
+import { useRangePicker } from '../composables/use-range-picker'
 import MonthTable from './basic-month-table.vue'
 
-import type { SetupContext } from 'vue'
 import type { Dayjs } from 'dayjs'
-import type { RangeState } from '../props/shared'
 
 defineOptions({
   name: 'DatePickerMonthRange',
 })
 
 const props = defineProps(panelMonthRangeProps)
 const emit = defineEmits(panelMonthRangeEmits)
-const attrs = useAttrs()
-const slots = useSlots()
 
-const { pickerNs: ppNs } = inject(ROOT_PICKER_INJECTION_KEY)!
-const drpNs = useNamespace('date-range-picker')
 const { lang } = useLocale()
+const {
+  minDate,
+  maxDate,
+  rangeState,
+  ppNs,
+  drpNs,
+
+  handleChangeRange,
+  handleRangeConfirm,
+  handleShortcutClick,
+  onSelect,
+} = useRangePicker()
```

Comment:
Maybe we can get `lang` from `useRangePicker`.

Reply:
We will fix that in an other commit.